tests/guestlog/guestlog.go

```diff
@@ -0,0 +1,249 @@
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+const cirrosStartupTimeout = 60
+const testString = "GuestConsoleTest3413254123535234523"
+
+var _ = Describe("[sig-compute]Guest console log", decorators.SigCompute, Serial, func() {
+
+	var (
+		virtClient kubecli.KubevirtClient
+		alpineVmi  *v1.VirtualMachineInstance
+		cirrosVmi  *v1.VirtualMachineInstance
+	)
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+
+		alpineVmi = libvmi.NewAlpine()
+		alpineVmi.Spec.Domain.Devices.AutoattachSerialConsole = pointer.P(true)
+		alpineVmi.Spec.Domain.Devices.LogSerialConsole = pointer.P(true)
+
+		cirrosVmi = libvmi.NewCirros()
+		cirrosVmi.Spec.Domain.Devices.AutoattachSerialConsole = pointer.P(true)
+		cirrosVmi.Spec.Domain.Devices.LogSerialConsole = pointer.P(true)
+	})
+
+	Describe("[level:component] Guest console log container", func() {
+		Context("set LogSerialConsole", func() {
+			DescribeTable("should successfully start with LogSerialConsole", func(autoattachSerialConsole, logSerialConsole, expected bool) {
+				By("Starting a VMI")
+				alpineVmi.Spec.Domain.Devices.AutoattachSerialConsole = pointer.P(autoattachSerialConsole)
+				alpineVmi.Spec.Domain.Devices.LogSerialConsole = pointer.P(logSerialConsole)
+				vmi := tests.RunVMIAndExpectLaunch(alpineVmi, cirrosStartupTimeout)
+
+				By("Finding virt-launcher pod")
+				virtlauncherPod, err := libvmi.GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
+				Expect(err).ToNot(HaveOccurred())
+				if expected {
+					Expect(virtlauncherPod.Spec.Containers).To(HaveLen(3))
+				} else {
+					Expect(virtlauncherPod.Spec.Containers).To(HaveLen(2))
+				}
+				foundContainer := false
+				for _, container := range virtlauncherPod.Spec.Containers {
+					if container.Name == "guest-console-log" {
+						foundContainer = true
+					}
+				}
+				Expect(foundContainer).To(Equal(expected))
+
+				if expected {
+					for _, containerStatus := range virtlauncherPod.Status.ContainerStatuses {
+						if containerStatus.Name == "guest-console-log" {
+							Expect(containerStatus.State.Running).ToNot(BeNil())
+						}
+					}
+				}
+			},
+				Entry("with AutoattachSerialConsole and LogSerialConsole", true, true, true),
+				Entry("with AutoattachSerialConsole but not LogSerialConsole", true, false, false),
+				Entry("without AutoattachSerialConsole but with LogSerialConsole", false, true, false),
+				Entry("without AutoattachSerialConsole and without LogSerialConsole", false, false, false),
+			)
+
+		})
+
+		Context("fetch logs", func() {
+			var k8sClient string
+			var cirrosLogo = strings.Replace(` ____               ____  ____
+ / __/ __ ____ ____ / __ \/ __/
+/ /__ / // __// __// /_/ /\ \ 
+\___//_//_/  /_/   \____/___/ 
+   http://cirros-cloud.net
+`, "\n", "\r\n", 5)
+
+			BeforeEach(func() {
+				k8sClient = clientcmd.GetK8sCmdClient()
+				clientcmd.SkipIfNoCmd(k8sClient)
```

Comment:
Is `SkipIfNoCmd` really necessary? Skipped tests can easily be overlooked.

Reply:
done